FlashLFQ/Bayesian Protein Quantification/ProteinQuantificationEngine.cs

```diff
@@ -0,0 +1,604 @@
+                    for (int i = partitionRange.Item1; i < partitionRange.Item2; i++)
+                    {
+                        ProteinGroup protein = proteinList[i].Key;
+
+                        // get the fold-change measurements for the peptides assigned to this protein
+                        List<double> peptideFoldChanges = GetPeptideFoldChangeMeasurements(protein, BaseCondition, condition, null, null);
+
+                        ProteinQuantificationEngineResult result;
+
+                        if (peptideFoldChanges.Count > 1)
+                        {
+                            // run the Bayesian analysis
+                            result = RunBayesianProteinQuant(peptideFoldChanges, protein, condition,
+                                randomSeedsForEachProtein[protein][0], null, false, BurnInSteps, NSteps, out var mus);
+
+                            var skepticalResult = RunBayesianProteinQuant(peptideFoldChanges, protein, condition,
+                                randomSeedsForEachProtein[protein][1], nullHyp, true, BurnInSteps, NSteps, out var skepticalMus);
+
+                            result.cutoff = nullHyp;
+                            result.CalculatePosteriorErrorProbability(skepticalMus);
+                        }
+                        else if (peptideFoldChanges.Count == 1)
+                        {
+                            result = new ProteinQuantificationEngineResult(protein, BaseCondition, condition,
+                                new double[] { peptideFoldChanges.First() }, new double[] { double.NaN },
+                                new double[] { double.NaN }, peptideFoldChanges, ProteinToBaseConditionIntensity[protein]);
+
+                            result.cutoff = nullHyp;
+                            result.CalculatePosteriorErrorProbability(new double[] { nullHyp });
+                        }
+                        else
+                        {
+                            result = new ProteinQuantificationEngineResult(protein, BaseCondition, condition,
+                                new double[] { 0.0 }, new double[] { double.NaN }, new double[] { double.NaN },
+                                peptideFoldChanges, ProteinToBaseConditionIntensity[protein]);
+
+                            result.cutoff = nullHyp;
+                            result.CalculatePosteriorErrorProbability(new double[] { nullHyp });
+                        }
+
+                        protein.conditionToQuantificationResults.Add(condition, result);
+                    }
+                });
+            }
+
+            // calculate FDR for the condition
+            foreach (var condition in conditions)
+            {
+                var res = proteinList.Select(p => p.Key.conditionToQuantificationResults[condition]).ToList();
+                CalculateFalseDiscoveryRates(res);
+            }
+
+            // remove temporary conditions
+            if (!conditionsAreDefined)
```

Comment:
So if a single file doesn't have a condition defined, then all files have their condition wiped?

Reply:
MM and FlashLFQ (the clients that call this) do not allow blank conditions, specifically for this reason

Reply:
...then why support it? If there's a blank condition, shouldn't you just return?

Reply:
I want to be able to do quantification without defining the conditions. but the code gets a lot messier if you have to check for conditions constantly. so it's just easier to set the conditions temporarily and then do the analysis, then undo the temporary conditions